MapleServer2/PacketHandlers/Game/QuestHandler.cs

```diff
@@ -106,7 +106,11 @@ private static void HandleCompleteQuest(GameSession session, PacketReader packet
         foreach (QuestRewardItem reward in questStatus.RewardItems)
         {
             Item newItem = new(reward.Code, reward.Count, reward.Rank);
-            if (newItem.RecommendJobs.Contains(session.Player.Job) || newItem.RecommendJobs.Contains(0))
+            List<int> limitJobRequirements = ItemMetadataStorage.GetMetadata(reward.Code).Limit.JobRequirements ?? new()
+            {
+                0
+            };
```

Comment:
feels like this may affect more than just quests. could we implement this in item parser instead?